zipkin-autoconfigure/storage-cassandra3/src/main/java/zipkin/autoconfigure/storage/cassandra3/brave/TracedSession.java

```diff
@@ -11,204 +11,182 @@
-    if (brave.serverSpanThreadBinder().getCurrentServerSpan() != null &&
-        brave.serverSpanThreadBinder().getCurrentServerSpan().getSpan() != null
-        && method.getName().equals("executeAsync") && args[0] instanceof BoundStatement) {
-      BoundStatement statement = (BoundStatement) args[0];
-
-      // via an internal class z.s.cassandra3.NamedBoundStatement, toString() is a nice name
-      SpanId spanId = brave.clientTracer().startNewSpan(statement.toString());
-
-      // o.a.c.tracing.Tracing.newSession must use the same format for the key zipkin
-      if (version.compareTo(ProtocolVersion.V4) >= 0) {
-        statement.enableTracing();
-        statement.setOutgoingPayload(singletonMap("zipkin", ByteBuffer.wrap(spanId.bytes())));
-      }
-
-      brave.clientTracer().setClientSent(); // start the span and store it
-      brave.clientTracer()
-          .submitBinaryAnnotation("cql.query", statement.preparedStatement().getQueryString());
-      cache.put(statement, brave.clientSpanThreadBinder().getCurrentClientSpan());
-      // let go of the client span as it is only used for the RPC (will have no local children)
-      brave.clientSpanThreadBinder().setCurrentSpan(null);
-      return new BraveResultSetFuture(target.executeAsync(statement), brave);
+    private static final Executor EXECUTOR = MoreExecutors.directExecutor();
+    private final Session session;
+    private final Brave brave;
+    private final SpanCollector collector;
+    private final ClientTracer clientTracer;
+    private final long useKeyspaceTimeout;
+    private volatile boolean keyspaceSet = false;
+    private final ProtocolVersion version;
+
+    static TracedSession create(final Session session, Brave brave, SpanCollector collector) {
+        return new TracedSession(session, brave, collector);
     }
-    try {
-      return method.invoke(target, args);
-    } catch (InvocationTargetException e) {
-      if (e.getCause() instanceof RuntimeException) throw e.getCause();
-      throw e;
+
+    private TracedSession(final Session session, Brave brave, SpanCollector collector) {
+        this.session = session;
+        this.brave = brave;
+        this.collector = collector;
+        this.clientTracer = brave.clientTracer();
+        this.useKeyspaceTimeout = session.getCluster().getConfiguration().getSocketOptions().getConnectTimeoutMillis();
+        this.version = session.getCluster().getConfiguration().getProtocolOptions().getProtocolVersion();
     }
-  }
-
-  @Override public void update(Host host, Statement statement, Exception e, long nanos) {
-    if (!(statement instanceof BoundStatement)) return;
-    Span span = cache.remove(statement);
-    if (span == null) {
-      if (statement.isTracing()) {
-        LOG.warn("{} not in the cache eventhough tracing is on", statement);
-      }
-      return;
+
+    @Override
+    public String getLoggedKeyspace() {
+        return session.getLoggedKeyspace();
```

Comment:
formatting is off. zipkin is 2 spaces, etc.  Easiest way (if using intellij) is to use this: https://github.com/square/java-code-styles

2 space tab and indent, with 4 space continuation indent... https://github.com/openzipkin/zipkin/blob/master/CONTRIBUTING.md